components/tabs/src/index.js

```diff
@@ -0,0 +1,164 @@
+ *     }));
+ *   }
+ *
+ *   handleClick(e, index) {
+ *     const mql = window.matchMedia(`(min-width: ${BREAKPOINTS.TABLET})`);
+ *     if (mql.matches) {
+ *       e.preventDefault();
+ *     }
+ *     return this.setTabIndex(index);
+ *   }
+ *
+ *   render() {
+ *     const { tabIndex } = this.state;
+ *     return (
+ *       <Tabs>
+ *         <Title>Content</Title>
+ *         <List>
+ *           <Tab
+ *             onClick={(event) => this.handleClick(event, 0)}
+ *             selected={tabIndex === 0}
+ *             href="#first-panel"
+ *           >
+ *             Title 1
+ *           </Tab>
+ *           <Tab
+ *             onClick={(event) => this.handleClick(event, 1)}
+ *             selected={tabIndex === 1}
+ *             href="#second-panel"
+ *           >
+ *             Title 2
+ *           </Tab>
+ *         </List>
+ *         <Panel selected={tabIndex === 0} id="first-panel">Panel content 1</Panel>
+ *         <Panel selected={tabIndex === 1} id="second-panel" >Panel content 2</Panel>
+ *       </Tabs>
+ *     );
+ *   }
+ * }
+ * ```
+ *
+ * ##### Hooks Example
+ * ```js
+ * const App = ({ defaultIndex}) => {
+ *  const [tabIndex, setTabIndex] = React.useState(defaultIndex);
+ *
+ *  const handleTabChange = newTabIndex => setTabIndex(newTabIndex);
+ *
+ *  function handleClick({ event: e, index }) {
+ *    const mql = window.matchMedia(`(min-width: ${BREAKPOINTS.TABLET})`);
+ *    if (mql.matches) {
+ *      e.preventDefault();
+ *    }
+ *    return handleTabChange(index);
+ *  }
+ *
+ *  return (
+ * <Tabs>
+ *   <Tabs.Title>Content</Tabs.Title>
+ *   <Tabs.List>
+ *     {[
+ *       {
+ *         content: 'Title 1',
+ *         href: '#first-panel',
+ *       },
+ *       {
+ *         content: 'Title 2',
+ *         href: '#second-panel',
+ *       },
+ *     ].map(({ content, href }, index) => (
+ *       <Tabs.Tab
+ *           onClick={(event) => handleClick({ event, index })}
+ *           selected={tabIndex === index}
+ *           href={href}
+ *         >
+ *           {content}
+ *         </Tab>
+ *       ))
+ *     }
+ *   </Tabs.List>
+ *     {[
+ *       {
+ *         content: 'Panel content 1',
+ *         id: 'first-panel',
+ *       },
+ *       {
+ *         content: 'Panel content 2',
+ *         id: 'second-panel',
+ *       },
+ *     ].map(({ content, id }, index) => (
+ *       <Tabs.Panel
+ *           selected={tabIndex === index}
+ *           id={id}
+ *         >
+ *           {content}
+ *         </Tabs.Panel>
+ *       ))
+ *     }
+ *    </Tabs>
+ *  );
+ * }
+ *
+ * ```
+ *
+ *
+ * ### References:
+ * - https://github.com/alphagov/govuk-frontend/tree/master/src/components/tabs
+ * - https://design-system.service.gov.uk/components/tabs/
+ *
+ */
+const DocumentedTabs = props => <Tabs {...props} />;
+
+Tabs.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+Tabs.Tab = Tab;
+Tabs.List = List;
+Tabs.Panel = Panel;
+Tabs.Title = Title;
+
+export { DocumentedTabs };
```

Comment:
as per @stevesims 's comment above, perhaps we also want to export Tab, List, Panel, Title here (as well as them being appended to component).

I'm not 100% but exporting in both ways may help people who are wrapping Tabs with an HOC and therefore lose Tabs.Tab (not a use case we've come across yet so hard to be sure)